source/h2_decoder.c

```diff
@@ -1034,12 +1050,24 @@ static struct aws_h2err s_state_fn_frame_goaway_debug_data(
 
     struct aws_byte_cursor debug_data = s_decoder_get_payload(decoder, input);
     if (debug_data.len > 0) {
-        DECODER_CALL_VTABLE_ARGS(decoder, on_goaway_i, debug_data);
+        if (decoder->goaway_in_progress.debug_data.len < decoder->goaway_in_progress.debug_data.capacity) {
+            /* Unless debug_data buffer is initialized, otherwise, ignore the debug data */
+            aws_byte_buf_append(&decoder->goaway_in_progress.debug_data, &debug_data);
+        }
```

Comment:
trivial: this if-check and comment are from the days where let the user set a max-size.

you can just blindly append now. maybe leave a note that we initialized the buffer to be exactly the right amount.